core/perf_test/CMakeLists.txt

```diff
@@ -114,3 +114,91 @@ IF(NOT Kokkos_ENABLE_OPENMPTARGET)
     CATEGORIES PERFORMANCE
   )
 ENDIF()
+
+
+IF(NOT Kokkos_ENABLE_BENCHMARKS OR KOKKOS_HAS_TRILINOS)
+  RETURN()
+ENDIF()
+
+find_package(benchmark QUIET)
+IF(NOT benchmark_FOUND)
+  message(STATUS "No installed benchmark found, fetching from GitHub")
+  include(FetchContent)
+  SET(BENCHMARK_ENABLE_TESTING OFF)
+
+  FetchContent_Declare(
+        googlebenchmark
+        GIT_REPOSITORY https://github.com/google/benchmark.git
+        GIT_TAG v1.6.2)
```

Comment:
Why not use the `ExternalProject_Add()` ? Or do we need the repo?

Reply:
I don't have a strong preference here.
- we have the lib available at configure stage, earlier feedback for any issues
- I haven't tried switching to `ExternalProject`, but I'd expect the code using `FetchContent` to be a bit more concise

Reply:
does it still configure if the user downloads the repo on another machine and copies it into the corresponding path in kokkos? Might be necessary for some machines without direct access to the internet. 
I am strongly in favor of letting cmake handle it and not just ship with a hard copy of the lib

Reply:
@no internet access: I'm trying to verify that using [FETCHCONTENT_FULLY_DISCONNECTED](https://cmake.org/cmake/help/latest/module/FetchContent.html#variable:FETCHCONTENT_FULLY_DISCONNECTED) option
@hard copy: current implementation is completely CMake based and there is no hard copy involved (this was addressed using `FetchContent`).

Reply:
I haven't found a reliable way to trick `FetchContent`, but building and installing `benchmark` locally and using `CMAKE_SYSTEM_PREFIX_PATH` works just fine (`find_package` will find the provided build).

Reply:
I was just thinking, that it would be cool if we would be able to not require `git` to be installed (as seen in the changes to the ci).

Reply:
Have you tried not using the `GIT_REPOSITORY` but just a `URL`? This would maybe rid of the dependency on git and allow to manually download a zip/tar and putting it in the right place